Test failure due to Podman version

**Describe the bug**

`TestHelloWorldCLIOnPodman/service-create-bind` fails on Podman version 5, with an error such as below:

    2024/12/11 11:42:06 service-create-bind has failed, exiting
    2024/12/11 11:42:06 STDOUT:
    Services exposed through Skupper:
    ├─ hello-world-backend:8080 (tcp)
    │  ╰─ Host ports:
    │     ╰─ ip: 0.0.0.0 - ports: 8081 -> 8080
    ╰─ hello-world-frontend:8080 (tcp)
       ╰─ Host ports:
          ╰─ ip: 0.0.0.0 - ports: 8080 -> 8080
    2024/12/11 11:42:06 STDERR:
        tester.go:161: Error: context deadline exceeded

On investigation, it was determined that the way Podman reports the `HostIp` field changed.

On version 4.x, when a container ports bound to all interfaces, on inspection its `HostIp` field would be empty:

               "PortBindings": {                                                                                                                                                                                                              
                    "8080/tcp": [                                                                                                                                                                                                             
                         {                                                                                                                                                                                                                    
                              "HostIp": "",                                                                            
                              "HostPort": "8080"                                                                                                                                                                                              
                         }                                                                                                                                                                                                                    
                    ]                                                                                                                                                                                                                         
               },                                                                                                                                                                                                                             

On version 5.x that changed; 

               "PortBindings": {
                    "8000/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "8001"
                         }
                    ]
               },


**How To Reproduce**

    $ skupper init --platform podman
    Skupper is now installed for user 'x'.  Use 'skupper status' to get more information.
    $ skupper --platform podman service create asdf 8000 --protocol tcp --container-name asdf --host-port 8000:8001

On the next steps, the result will be different between Podman 4 and 5.

Podman 4.9.4:

    $ skupper --platform podman service status
    Services exposed through Skupper:
    ╰─ asdf:8000 (tcp)
       ╰─ Host ports:
          ╰─ ip: * - ports: 8001 -> 8000
    $ podman container inspect asdf | grep HostIp
                              "HostIp": "",
                              "HostIp": "",

Podman 5.2.2:

    $ skupper --platform podman service status
    Services exposed through Skupper:
    ╰─ asdf:8000 (tcp)
       ╰─ Host ports:
          ╰─ ip: 0.0.0.0 - ports: 8001 -> 8000
    $ podman container inspect asdf | grep HostIp
                                  "HostIp": "0.0.0.0",
                                  "HostIp": "0.0.0.0",
    
The way `service. StatusTester` works, it expected either that literal asterisk, or a specific host configured on the test table.

The Podman change reflected on the Skupper output, and both are only informational.  We did not observe any impacts on skupper operation itself.

**Expected behavior**

The test should pass on Podman 5.

**Environment details**
 - Skupper CLI:1.5.5, 1.4.X
 - Skupper Operator (if applicable): N/A
 - Platform: podman

**Additional context**
N/A

--- a/test/utils/k8s/job.go
+++ b/test/utils/k8s/job.go
@@ -175,19 +175,29 @@ func WaitForJob(ns string, kubeClient kubernetes.Interface, jobName string, time
 		case <-timeoutCh:
 			return nil, fmt.Errorf("Timeout: Job is still active: %s", jobName)
 		case <-tick:
-			job, _ := jobsClient.Get(context.TODO(), jobName, metav1.GetOptions{})
+			job, err := jobsClient.Get(context.TODO(), jobName, metav1.GetOptions{})
+
+			if err != nil {
+				return nil, err // Handle the error
+			}
 
 			if job.Status.Active > 0 {
 				fmt.Println("Job is still active")
 			} else if len(job.Status.Conditions) > 0 {
-				if job.Status.Conditions[0].Type == batchv1.JobComplete {
-					fmt.Println("Job Successful!")
-					return job, nil
-				} else if job.Status.Conditions[0].Type == batchv1.JobFailed {
-					statusJson, _ := json.Marshal(job.Status)
-					fmt.Printf("Job failed?, status = %v\n", string(statusJson))
-					return job, fmt.Errorf("Job failed. Status: %s", string(statusJson))
+				for _, condition := range job.Status.Conditions {
+					switch condition.Type {
+					case batchv1.JobComplete:
+						fmt.Println("Job Successful!")
+						return job, nil
+					case batchv1.JobFailed:
+						statusJson, _ := json.Marshal(job.Status)
+						fmt.Printf("Job failed?, status = %v\n", string(statusJson))
+						return job, fmt.Errorf("Job failed. Status: %s", string(statusJson))
+					}
 				}
+				// No terminal condition detected above
+				statusJson, _ := json.Marshal(job.Status)
+				fmt.Printf("Job is not active, but it is neither successful nor failed.  Status:\n%v\n", string(statusJson))
 			} else {
 				fmt.Println("Waiting on job condition")
 			}

--- a/test/utils/skupper/cli/service/status.go
+++ b/test/utils/skupper/cli/service/status.go
@@ -111,7 +111,7 @@ func (s *StatusTester) run(platform types.Platform, cluster *base.ClusterContext
 	for _, svc := range s.ServiceInterfaces {
 		serviceEntry := fmt.Sprintf(`.*%s:%d \(%s\)`, svc.Address, svc.Ports[0], svc.Protocol)
 		if hostPortBinding, ok := s.Podman.GetHostPortBinding(svc.Address); ok {
-			hostIp := utils.DefaultStr(hostPortBinding.HostIp, `\*`)
+			hostIp := utils.DefaultStr(hostPortBinding.HostIp, `(\*|0\.0\.0\.0)`)
 			var portMapping string
 			var portMappingPrefix string
 			for svcPort, hostPort := range hostPortBinding.HostPorts {

